aries_vcx/src/common/ledger/transactions.rs

```diff
@@ -124,6 +125,46 @@ pub async fn into_did_doc(profile: &Arc<dyn Profile>, invitation: &Invitation) -
     Ok(did_doc)
 }
 
+fn _ed25519_public_key_to_did_key(public_key_base58: &str) -> VcxResult<String> {
+    let public_key_bytes = bs58::decode(public_key_base58).into_vec().unwrap();
+    // Add the multicodec prefix for ed25519 (0xed 0x01)
+    let mut did_key_bytes = vec![0xed, 0x01];
+    did_key_bytes.extend_from_slice(&public_key_bytes);
+    // Base58 encode the resulting bytes and add the "z" prefix for base58 encoding
+    let mut did_key = String::from("z");
+    did_key.push_str(&bs58::encode(&did_key_bytes).into_string());
+    // Add the "did:key:" prefix
+    did_key.insert_str(0, "did:key:");
+    Ok(did_key)
+}
+
+fn did_key_to_public_key(recipient_keys: Vec<String>) -> Vec<String> {
+    let did_key_prefix = "did:key:";
+
+    let mut result = Vec::new();
+    let edd2599 = [0xed, 0x01];
+
+    for recipient_key in recipient_keys {
+        if recipient_key.starts_with(did_key_prefix) {
+            let fingerprint = recipient_key[did_key_prefix.len()..].to_string();
+
+            let fingerprint = match fingerprint.chars().nth(0) {
+                Some('z') => &fingerprint[1..],
+                _ => &fingerprint,
+            };
+
+            let decoded_value = bs58::decode(fingerprint).into_vec().unwrap();
```

Comment:
preferably no unwraps, as this could panic the program - perhaps map the error result to the `InvalidDid` error and return early. Something like:

```
            let decoded_value = bs58::decode(fingerprint).into_vec().map_err(|_| {
                VcxError::from_msg(
                    VcxErrorKind::InvalidDid,
                    format!("Could not base58 decode a did:key fingerprint: {}", recipient_key),
                )
            })?;
```